src/frontend/src/flows/addDevice/welcomeView/registerTentativeDevice.ts

```diff
@@ -1,174 +1,168 @@
  * @param userNumber anchor to add the tentative device to.
  */
-export const registerTentativeDevice = (
+export const registerTentativeDevice = async (
   userNumber: bigint,
   connection: Connection
 ): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent({ userNumber }), container);
-  return init(userNumber, connection);
-};
+  // First, we need an alias for the device to (tentatively) add
+  const alias = await promptDeviceAlias({
+    title: "Add a Trusted Device",
+    message: html` What device do you want to add to anchor
+      <strong class="t-strong">${userNumber}</strong>?`,
+  });
 
-export type TentativeDeviceInfo = [
-  bigint,
-  string,
-  KeyType,
-  Purpose,
-  DerEncodedPublicKey,
-  ArrayBuffer
-];
+  if (alias === null) {
+    // TODO L2-309: do this without reload
+    window.location.reload();
+    return;
+  }
 
-export const addTentativeDevice = async (
-  connection: Connection,
-  tentativeDeviceInfo: TentativeDeviceInfo
-): Promise<void> => {
+  // Then, we create local WebAuthn credentials for the device
   const result = await withLoader(() =>
-    connection.addTentativeDevice(...tentativeDeviceInfo)
+    createDevice({ userNumber, connection })
   );
 
-  if ("added_tentatively" in result) {
-    await showVerificationCode(
-      tentativeDeviceInfo[0],
-      connection,
-      tentativeDeviceInfo[1],
-      result.added_tentatively,
-      Array.from(new Uint8Array(tentativeDeviceInfo[5]))
-    );
-  } else if ("device_registration_mode_off" in result) {
-    await deviceRegistrationDisabledInfo(connection, tentativeDeviceInfo);
-  } else if ("another_device_tentatively_added" in result) {
+  if (result instanceof Error) {
     await displayError({
-      title: "Tentative Device Already Exists",
-      message:
-        'The "add device" process was already started for another device. If you want to add this device instead, log in using an existing device and restart the "add device" process.',
+      title: "Error adding new device",
+      message: "Unable to register new WebAuthn Device.",
+      detail: result.message,
       primaryButton: "Ok",
     });
     // TODO L2-309: do this without reload
-    window.location.reload();
-  } else {
-    throw new Error(
-      "unknown tentative device registration result: " + JSON.stringify(result)
-    );
+    return window.location.reload() as never;
   }
-};
 
-const init = async (userNumber: bigint, connection: Connection) => {
-  const existingAuthenticators = await withLoader(() =>
-    connection.lookupAuthenticators(userNumber)
-  );
-  const cancelButton = document.getElementById(
-    "registerTentativeDeviceCancel"
-  ) as HTMLButtonElement;
-
-  cancelButton.onclick = () => {
-    // TODO L2-309: do this without reload
-    window.location.reload();
-  };
+  // Finally, we submit it to the canister
+  const device: Omit<DeviceData, "origin"> & { credential_id: [CredentialId] } =
```

Comment:
did not checked this in details, was refactored without any changes right?

Reply:
The values are the same, but it's now an object instead of a big tuple (`[bigint, string, ...]`)